packages/webpack-cli/lib/groups/resolveMode.js

```diff
@@ -0,0 +1,25 @@
+const PRODUCTION = 'production';
+const DEVELOPMENT = 'development';
+
+const resolveMode = (args) => {
+    const {
+        env: { NODE_ENV },
+    } = process;
+    const { mode } = args;
+    let finalMode;
+    /**
+     * It determines the mode to pass to webpack compiler
+     * @returns {string} The mode
+     */
+    if (NODE_ENV && (NODE_ENV === PRODUCTION || NODE_ENV === DEVELOPMENT)) {
+        finalMode = NODE_ENV;
+    } else if (mode) {
+        finalMode = mode;
+    }
```

Comment:
Here bug :smile: 

we always set `finalMode` to `NODE_ENV`, even you specify `mode`, it is big bug

```js
if (!mode && NODE_ENV && (NODE_ENV === PRODUCTION || NODE_ENV === DEVELOPMENT)) {
   finalMode = NODE_ENV;
}
```

Reply:
Damn, it has been this way before, good that we caught this during refactor. We should have a test for this 😄

Reply:
Can you fix it here?

Reply:
yes wip 😄

Reply:
Ah so `mode` has a default value set to `production` hence !mode condition is not relevant here so its this way

Reply:
Hence supplied `NODE_ENV` takes precedence

Reply:
Right thanks for confirming, fixed 💯

Reply:
Does `args` contain only arguments? Because we should respect the value from config and do not override it

Reply:
We already respect, added test 👍